packages/@aws-cdk/aws-appmesh/test/test.route.ts

```diff
@@ -812,6 +812,65 @@ export = {
     test.done();
   },
 
+  'should throw an error with invalid number of headers'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const mesh = new appmesh.Mesh(stack, 'mesh', {
+      meshName: 'test-mesh',
+    });
+
+    const router = new appmesh.VirtualRouter(stack, 'router', {
+      mesh,
+    });
+
+    const virtualNode = mesh.addVirtualNode('test-node', {
+      serviceDiscovery: appmesh.ServiceDiscovery.dns('test'),
+      listeners: [appmesh.VirtualNodeListener.http()],
+    });
+
+    // WHEN + THEN
+    test.throws(() => {
+      router.addRoute('route', {
+        routeSpec: appmesh.RouteSpec.http2({
+          weightedTargets: [{ virtualNode }],
+          match: {
+            prefixPath: '/',
+            // Empty header
+            headers: [],
+          },
+        }),
+      });
+    });
+
+    test.throws(() => {
```

Comment:
Same here - please add the expected message to the `throws()` assertion (otherwise, the test might succeed because of throwing a completely unrelated exception).

Reply:
Added